src/combat.ts

```diff
@@ -11,17 +11,30 @@ export default class Macro extends StrictMacro {
         `!mpbelow ${mpCost($skill`Stuffed Mortar Shell`)}`,
         Macro.trySkill($skill`Stuffed Mortar Shell`)
       );
-    return (useCinch ? macroHead.trySkill($skill`Cincho: Confetti Extravaganza`) : macroHead)
-      .while_(
-        `!mpbelow ${mpCost($skill`Saucegeyser`)} && hasskill ${toInt($skill`Saucegeyser`)}`,
-        Macro.skill($skill`Saucegeyser`)
-      )
-      .while_(
-        `!mpbelow ${mpCost($skill`Saucestorm`)} && hasskill ${toInt($skill`Saucestorm`)}`,
-        Macro.skill($skill`Saucestorm`)
-      )
-      .attack()
-      .repeat();
+    if (!mainStat === $stat`Muscle`) {
+      //Update to add alternative combat options for non-Mys classes
+      return (useCinch ? macroHead.trySkill($skill`Cincho: Confetti Extravaganza`) : macroHead)
+        .while_(
+          `!mpbelow ${mpCost($skill`Saucegeyser`)} && hasskill ${toInt($skill`Saucegeyser`)}`,
+          Macro.skill($skill`Saucegeyser`)
+        )
+        .while_(
+          `!mpbelow ${mpCost($skill`Saucestorm`)} && hasskill ${toInt($skill`Saucestorm`)}`,
+          Macro.skill($skill`Saucestorm`)
+        )
+        .attack()
+        .repeat();
+    } else {
+      return (useCinch ? macroHead.trySkill($skill`Cincho: Confetti Extravaganza`) : macroHead)
+        .while_(
+          `!mpbelow ${mpCost($skill`Lunging Thrust-Smack`)} && hasskill ${toInt(
+            $skill`Lunging Thrust-Smack`
+          )}`,
+          Macro.skill($skill`Lunging Thrust-Smack`)
+        )
+        .attack()
+        .repeat();
+    }
```

Comment:
Use `if (A !== B)` instead of `if (!(A === B))`. In fact, you're using `if (!A === B)` which I think has a completely different behaviour

Reply:
Also, as previously mentioned, the macro logic goes
```
cinchLogic().startOfFightLogic().classSpecificAttackLogic().attack().repeat()
```
This should be implemented as such for easier reading, rather than repeating the exact same structure in both the `if` and `else` portion

Reply:
I've been running my head up against this for a while and can't figure it out; specifically the macro portion. I'll keep working on it.

Reply:
I couldn't figure this out; so Saucestorm for all.

Reply:
Perhaps something like this
```ts
kill(useCinch = false): Macro {
    const macroHead = this.trySkill($skill`Curse of Weaksauce`)
      .trySkill($skill`Sing Along`)
      .if_(
        `!mpbelow ${mpCost($skill`Stuffed Mortar Shell`)}`,
        Macro.trySkill($skill`Stuffed Mortar Shell`)
      );

    const whileCondition = (sk: Skill): [string, Macro] => {
      return [`!mpbelow ${sk} && hasskill ${toInt(sk)}`, Macro.skill(sk)];
    };

    return (useCinch ? macroHead.trySkill($skill`Cincho: Confetti Extravaganza`) : macroHead)
      .externalIf(
        mainStat === $stat`Muscle`,
        Macro.while_(...whileCondition($skill`Lunging Thrust-Smack`)),
        Macro.while_(...whileCondition($skill`Saucegeyser`)).while_(
          ...whileCondition($skill`Saucestorm`)
        )
      )
      .attack()
      .repeat();
  }
```

Reply:
This may end up being a point of preference. For lower shiny people, LTS will consume less MP and be more accessible. For higher shiny, the extra +mus is going to make saucegeyser preferable; and those folks typically have the MP to spare.... but they should also maybe look at other scripts.

So, would you prefer we opt to benefit lower shiny folks and target LTS?

Reply:
[InstantHCCS](https://github.com/Pantocyclus/InstantHCCS/blob/9fd8f3a871ff96548521d3af6e98d2dbbc3e5e27/src/combat.ts#L26) runs LTS just fine. I don't really see the issue here re: high vs low shiny